phoenix-core/src/it/java/org/apache/phoenix/end2end/RowValueConstructorOffsetIT.java

```diff
@@ -0,0 +1,860 @@
+            assertEquals(2, k1);
+            assertEquals(3, k2);
+            assertEquals(3, k3);
+        }
+        assertFalse(rs.next());
+    }
+
+    @Test
+    public void testBindsRVCOffsetLookup() throws SQLException {
+        String sql = "SELECT * FROM " + dataTableName + "  LIMIT 2 OFFSET (k1,k2,k3)=(?, ?, ?)";
+        PreparedStatement ps = conn.prepareStatement(sql);
+
+        ps.setInt(1,2);
+        ps.setInt(2,3);
+        ps.setInt(3,2);
+        ResultSet rs = ps.executeQuery();
+
+        assertTrue(rs.next());
+        {
+            int k1 = rs.getInt(1);
+            int k2 = rs.getInt(2);
+            int k3 = rs.getInt(3);
+            assertEquals(2, k1);
+            assertEquals(3, k2);
+            assertEquals(2, k3);
+        }
+        assertTrue(rs.next());
+        {
+            int k1 = rs.getInt(1);
+            int k2 = rs.getInt(2);
+            int k3 = rs.getInt(3);
+            assertEquals(2, k1);
+            assertEquals(3, k2);
+            assertEquals(3, k3);
+        }
+        assertFalse(rs.next());
+
+        //TODO try index bind path
+    }
+
+
+
+    // Test RVC Offset where clause
+    @Test
+    public void testWhereClauseRVCOffsetLookup() throws SQLException {
+       //Offset should not overcome the where clause
+        String sql = "SELECT * FROM " + dataTableName + " WHERE (k1,k2,k3)=(3,3,3) LIMIT 2 OFFSET (k1,k2,k3)=(2, 3, 1)";
+        ResultSet rs = conn.createStatement().executeQuery(sql);
+        assertTrue(rs.next());
+        {
+            int k1 = rs.getInt(1);
+            int k2 = rs.getInt(2);
+            int k3 = rs.getInt(3);
+            assertEquals(3, k1);
+            assertEquals(3, k2);
+            assertEquals(3, k3);
+        }
+        assertFalse(rs.next());
+    }
+
+    @Test
+    public void testSaltedTableRVCOffsetOrderBy() throws SQLException {
+        //Make a salted table
+        String saltedTableName = "T_" + generateUniqueName();
+
+        //
+        String saltedDDL = String.format(DATA_DDL + "SALT_BUCKETS=4",saltedTableName);
+
+        conn.createStatement().execute(saltedDDL);
+        conn.commit();
+
+        //If we attempt to order by the row key we should not fail
+        boolean failed = false;
+        String sql = "SELECT * FROM " + saltedTableName + " ORDER BY K1,K2,K3 LIMIT 2 OFFSET (k1,k2,k3)=(2, 3, 1)";
+        try {
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+        } catch(Exception e) {
+            //Should fail
+            failed = true;
+        }
+        assertFalse(failed);
+
+        //If we attempt to order by the not row key we should fail
+        failed = false;
+        sql = "SELECT * FROM " + saltedTableName + " ORDER BY K2,K1,K3 LIMIT 2 OFFSET (k1,k2,k3)=(2, 3, 1)";
+        try {
+            ResultSet rs = conn.createStatement().executeQuery(sql);
+        } catch(Exception e) {
+            //Should fail
+            failed = true;
+        }
+        assertTrue(failed);
```

Comment:
little be confused by the naming, maybe `status` better nameing here?

Reply:
No longer applicable.